transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy

```diff
@@ -283,6 +285,141 @@ class CopySpec extends Specification {
+        def expectedConceptPaths2 = readFieldsFromFile(studyPart2Folder, Concepts.TABLE, 'concept_path')
+        def expectedModifierPaths2 = readFieldsFromFile(studyPart2Folder, Modifiers.TABLE, 'modifier_path')
+        def expectedSubjectIds2 =
+                readFieldsFromFile(studyPart2Folder, Patients.PATIENT_MAPPING_TABLE, 'patient_ide')
+        def expectedTreeNodePaths2 = readFieldsFromFile(studyPart2Folder, TreeNodes.TABLE, 'c_fullname')
+
+        CommandLine cli2 = new DefaultParser()
+                .parse(options, ['--incremental', '--directory', studyPart2Folder] as String[])
+        Copy.runCopy(cli2, DATABASE_CREDENTIALS)
+
+        Map countsAfterUpload2 = count(studySpecificTables)
+        def inDbPatientNums2 = readFieldsFromDb(Observations.TABLE,'patient_num') as Set
+        def inDbConceptPaths2 = readFieldsFromDb(Concepts.TABLE, 'concept_path')
+        def inDbModifierPaths2 = readFieldsFromDb(Modifiers.TABLE, 'modifier_path')
+        def inDbSubjectIds2 = readFieldsFromDb(Patients.PATIENT_MAPPING_TABLE, 'patient_ide')
+        def inDbPatientIdeToPatientNum = readPatientIdeToPatientNum()
+        def inDbTreeNodePaths2 = readFieldsFromDb(TreeNodes.TABLE, 'c_fullname')
+
+
+        then: 'the second part of the study has been loaded incrementally'
+        incrementalStudyDbIdentifier
+        /** 1 patient not present in part2: SURVEY_INC_P2 -> 3 observations
+         *  1 patient updated in part2    : SURVEY_INC_P1 -> 3 observations
+         *  1 patient added in part 2     : SURVEY_INC_P3 -> 3 observations
+         */
+        inDbPatientNums2.size() == inDbPatientNums1.size() + 1
+        countsAfterUpload2[Observations.TABLE] == countsAfterUpload1[Observations.TABLE] + 3
+        countsAfterUpload2[Studies.TRIAL_VISIT_TABLE] == countsAfterUpload1[Studies.TRIAL_VISIT_TABLE] + 1
+        countsAfterUpload2[Studies.STUDY_TABLE] == countsAfterUpload1[Studies.STUDY_TABLE]
+        countsAfterUpload2[Studies.STUDY_DIMENSIONS_TABLE] == countsAfterUpload1[Studies.STUDY_DIMENSIONS_TABLE]
+        inDbConceptPaths2.containsAll(expectedConceptPaths2)
+        inDbModifierPaths2.containsAll(expectedModifierPaths2)
+        inDbSubjectIds2.containsAll(expectedSubjectIds2)
+        inDbTreeNodePaths2.containsAll(expectedTreeNodePaths2)
+        inDbSubjectIds2.findAll{ it.toString().startsWith("SURVEY_INC_")} as Set ==
+                (expectedSubjectIds1 + expectedSubjectIds2) as Set
+
+        // observation for patient from part1 only - old value
+        readFieldsFromDb(Observations.TABLE,'nval_num',
+                "where patient_num=${inDbPatientIdeToPatientNum['SURVEY_INC_P2']} AND concept_cd='age'")[0] == 33
+        // observation for patient updated in part2 - updated value
+        readFieldsFromDb(Observations.TABLE,'nval_num',
+                "where patient_num=${inDbPatientIdeToPatientNum['SURVEY_INC_P1']} AND concept_cd='age'")[0] == 26
```

Comment:
It may be more convenient to test if `readFieldsFromDb(...) == [26]` and it also tests the number of observations for the patient.